lms/djangoapps/ccx/tests/test_models.py

```diff
@@ -185,12 +218,44 @@ def test_end_datetime_short_date(self):
     }))
     def test_end_datetime_date_time_format(self):
         """verify that the DATE_TIME format also works as expected"""
-        end = datetime(2015, 1, 1, 12, 0, 0, tzinfo=UTC())
+        end = datetime(2015, 1, 1, 12, 0, 0, tzinfo=utc)
         expected = "Jan 01, 2015 at 12:00 UTC"
         self.set_ccx_override('due', end)
         actual = self.ccx.end_datetime_text('DATE_TIME')  # pylint: disable=no-member
         self.assertEqual(expected, actual)
 
+    @freeze_time("2015-11-01 08:59:00")
+    def test_end_datetime_time_zone_before(self):
+        """
+        verify that end date is correctly converted when time zone specified
+        and before daylight savings switch
+        """
+        time_zone = timezone('America/Los_Angeles')
+        end = datetime(2015, 11, 1, 8, 59, 0, tzinfo=utc)
+        expected_short_date = "Nov 01, 2015"
+        expected_datetime = "Nov 01, 2015 at 01:59 PDT"
+        self.set_ccx_override('due', end)
+        actual_short_date = self.ccx.end_datetime_text(time_zone=time_zone)  # pylint: disable=no-member
+        actual_datetime = self.ccx.end_datetime_text('DATE_TIME', time_zone)  # pylint: disable=no-member
+        self.assertEqual(expected_short_date, actual_short_date)
+        self.assertEqual(expected_datetime, actual_datetime)
+
+    @freeze_time("2015-11-01 09:00:00")
+    def test_end_datetime_zone_after(self):
+        """
+        verify that end date is correctly converted when time zone specified
+        and after daylight savings switch
+        """
+        time_zone = timezone('America/Los_Angeles')
+        end = datetime(2015, 11, 1, 9, 00, 0, tzinfo=utc)
+        expected_short_date = "Nov 01, 2015"
+        expected_datetime = "Nov 01, 2015 at 01:00 PST"
+        self.set_ccx_override('due', end)
+        actual_short_date = self.ccx.end_datetime_text(time_zone=time_zone)  # pylint: disable=no-member
+        actual_datetime = self.ccx.end_datetime_text('DATE_TIME', time_zone)  # pylint: disable=no-member
+        self.assertEqual(expected_short_date, actual_short_date)
+        self.assertEqual(expected_datetime, actual_datetime)
```

Comment:
General note: it looks like all your `test_<some_date>_(before|after)` tests are similar - could you just write 1 test and use ddt to check both before and after?